### PR TITLE
FP-1234: Add Missing Top Margin for Headings

### PR DESCRIPTION
**Pull request by tacc-wbomar**
# To Do

- [x] Test on UTRC **after** 2021-09-27 launch.

# Required By

- https://github.com/TACC/Core-CMS/pull/360
  Which makes the missing top margin obvious for `<h1>` (which is used—rightly so—by UTRC).

# Overview

- Use (browser) default margin-top values for headings.
- Thus support a `margin-top` for `<h1>`'s (and other headings on all pages, consistently).

# Issues

- [CEPWMA-672](https://jira.tacc.utexas.edu/browse/CEPWMA-672)

# Screenshots

| `<h1>` | Typography | Guide | Standard | Full Width |
| :-: | :-: | :-: | :-: | :-: |
| UTRC H1 Test | Typography (MIgrate Styles Removed) | Guide Page (Happens to be the Typography page with template) | Skipped | Skipped |

# Testing

All headings\* on any UTRC page use default browser `margin-top`:
- ~~Style Guide (template)†~~‡
- Typography (snippet, `snippets/typography.html`)
- Guide(: …) (template)†
- Full Width (template)†
- Standard (template)†

> \* Headings are `<h1>`, `<h2>`, `<h3>`, `<h4>`, `<h5>`, `<h6>`.
> † To be available, it must be in `CMS_TEMPLATES`.
> ‡ This guide has its won margin. The Style Guide is private, not well-known, manual, and subject to change. Ignoring for now.

# Notes

<details><summary>Why was this not done during v1→v2 CMS migration?</summary>

@tacc-wbomar was lazy, because he was in a hurry **and** was hesitant because he did not have standard typography to work from. He previously did not use `<h1>`'s as first heading level. He used `<h2>` because the text was not as large, so the missing margin top was not as ugly. He should have used `<h1>`.

_@tacc-wbomar has since been given the reins by Design to standard typography._

</details>

### CODE REVIEW

File: taccsite_cms/static/site_cms/css/src/_imports/elements/html-elements.css [hunk 1]
```diff
@@ -63,6 +63,8 @@ main {
 }
 
 h1, h2, h3, h4, h5, h6 {
+  margin-top: revert; /* to overwrite Bootstrap `_reboot.css` */
```
Comment (tacc-wbomar) [inline taccsite_cms/static/site_cms/css/src/_imports/elements/html-elements.css:66]: **This is they _KEY_ change.** It undoes Bootstrap's `margin-top: 0`, and lets browser defaults apply.

File: taccsite_cms/static/site_cms/css/src/_imports/trumps/s-document.css [hunk 2]
```diff
@@ -20,19 +20,13 @@ Styleguide Trumps.Scopes.Document
...
 .s-document h3 {
   margin-bottom: var(--global-space--normal);
 }
 
-/* Extra space above lesser headings */
-.s-document h3 { margin-top: 1em; }
-.s-document h4,
-.s-document h5,
-.s-document h6 { margin-top: 1.25em; }
-
```
Comment (tacc-wbomar) [inline taccsite_cms/static/site_cms/css/src/_imports/trumps/s-document.css:35]: **This is a _supporting_ change.** it ensures Guide pages use the same margin-top for headings as other pages.

**Review by tacc-wbomar — COMMENTED** [review]
I added notes for reviewers.

**Review by taoteg — APPROVED** [review]
LGTM

**Comment by tacc-wbomar** [comment]
I should test this thoroughly first before merging. Good candidates:
- TUP Dev
- UTRC Pre-Prod

**Comment by tacc-wbomar** [comment]
Tested with UTRC.

**Pull request merged**